include/dxc/DxilContainer/RDAT_LibraryTypes.inl

```diff
@@ -65,175 +36,65 @@ RDAT_ENUM_END()
-RDAT_ENUM_VALUE_NODEF(StructuredBuffer)
-RDAT_ENUM_VALUE_NODEF(CBuffer)
-RDAT_ENUM_VALUE_NODEF(Sampler)
-RDAT_ENUM_VALUE_NODEF(TBuffer)
-RDAT_ENUM_VALUE_NODEF(RTAccelerationStructure)
-RDAT_ENUM_VALUE_NODEF(FeedbackTexture2D)
-RDAT_ENUM_VALUE_NODEF(FeedbackTexture2DArray)
-RDAT_ENUM_VALUE_NODEF(NumEntries)
-#if DEF_RDAT_ENUMS == DEF_RDAT_DUMP_IMPL
-static_assert((unsigned)hlsl::DXIL::ResourceKind::NumEntries == 19,
-              "otherwise, RDAT_DXIL_ENUM definition needs updating");
-#endif
-RDAT_ENUM_END()
-
-RDAT_DXIL_ENUM_START(hlsl::DXIL::ShaderKind, uint32_t)
-RDAT_ENUM_VALUE_NODEF(Pixel)
-RDAT_ENUM_VALUE_NODEF(Vertex)
-RDAT_ENUM_VALUE_NODEF(Geometry)
-RDAT_ENUM_VALUE_NODEF(Hull)
-RDAT_ENUM_VALUE_NODEF(Domain)
-RDAT_ENUM_VALUE_NODEF(Compute)
-RDAT_ENUM_VALUE_NODEF(Library)
-RDAT_ENUM_VALUE_NODEF(RayGeneration)
-RDAT_ENUM_VALUE_NODEF(Intersection)
-RDAT_ENUM_VALUE_NODEF(AnyHit)
-RDAT_ENUM_VALUE_NODEF(ClosestHit)
-RDAT_ENUM_VALUE_NODEF(Miss)
-RDAT_ENUM_VALUE_NODEF(Callable)
-RDAT_ENUM_VALUE_NODEF(Mesh)
-RDAT_ENUM_VALUE_NODEF(Amplification)
-RDAT_ENUM_VALUE_NODEF(Node)
-RDAT_ENUM_VALUE_NODEF(Invalid)
-#if DEF_RDAT_ENUMS == DEF_RDAT_DUMP_IMPL
-static_assert((unsigned)hlsl::DXIL::ShaderKind::Invalid == 16,
-              "otherwise, RDAT_DXIL_ENUM definition needs updating");
-#endif
-RDAT_ENUM_END()
-
-RDAT_DXIL_ENUM_START(hlsl::DXIL::SemanticKind, uint32_t)
-// clang-format off
-  // Python lines need to be not formatted.
-  /* <py::lines('SemanticKind-ENUM')>hctdb_instrhelp.get_rdat_enum_decl("SemanticKind", nodef=True)</py>*/
-  // clang-format onm
-  // SemanticKind-ENUM:BEGIN
-  RDAT_ENUM_VALUE_NODEF(Arbitrary)
-  RDAT_ENUM_VALUE_NODEF(VertexID)
-  RDAT_ENUM_VALUE_NODEF(InstanceID)
-  RDAT_ENUM_VALUE_NODEF(Position)
-  RDAT_ENUM_VALUE_NODEF(RenderTargetArrayIndex)
-  RDAT_ENUM_VALUE_NODEF(ViewPortArrayIndex)
-  RDAT_ENUM_VALUE_NODEF(ClipDistance)
-  RDAT_ENUM_VALUE_NODEF(CullDistance)
-  RDAT_ENUM_VALUE_NODEF(OutputControlPointID)
-  RDAT_ENUM_VALUE_NODEF(DomainLocation)
-  RDAT_ENUM_VALUE_NODEF(PrimitiveID)
-  RDAT_ENUM_VALUE_NODEF(GSInstanceID)
-  RDAT_ENUM_VALUE_NODEF(SampleIndex)
-  RDAT_ENUM_VALUE_NODEF(IsFrontFace)
-  RDAT_ENUM_VALUE_NODEF(Coverage)
-  RDAT_ENUM_VALUE_NODEF(InnerCoverage)
-  RDAT_ENUM_VALUE_NODEF(Target)
-  RDAT_ENUM_VALUE_NODEF(Depth)
-  RDAT_ENUM_VALUE_NODEF(DepthLessEqual)
-  RDAT_ENUM_VALUE_NODEF(DepthGreaterEqual)
-  RDAT_ENUM_VALUE_NODEF(StencilRef)
-  RDAT_ENUM_VALUE_NODEF(DispatchThreadID)
-  RDAT_ENUM_VALUE_NODEF(GroupID)
-  RDAT_ENUM_VALUE_NODEF(GroupIndex)
-  RDAT_ENUM_VALUE_NODEF(GroupThreadID)
-  RDAT_ENUM_VALUE_NODEF(TessFactor)
-  RDAT_ENUM_VALUE_NODEF(InsideTessFactor)
-  RDAT_ENUM_VALUE_NODEF(ViewID)
-  RDAT_ENUM_VALUE_NODEF(Barycentrics)
-  RDAT_ENUM_VALUE_NODEF(ShadingRate)
-  RDAT_ENUM_VALUE_NODEF(CullPrimitive)
-  RDAT_ENUM_VALUE_NODEF(Invalid)
-  // SemanticKind-ENUM:END
-RDAT_ENUM_END()
-
-RDAT_DXIL_ENUM_START(hlsl::DXIL::ComponentType, uint32_t)
   RDAT_ENUM_VALUE_NODEF(Invalid)
```

Comment:
I think this one mistaken match makes the diff actively unhelpful in trying to review this. If only the whitespace change could be separated from the rest, I think this would be a lot easier to review. I know it's some additional work, but I think it needs to be done in order to review this already tricky file. I'm inclined to do it myself. Is that okay?

Reply:
This is also a problem because I reordered definitions.  I can try to separate changes into three phases: whitespace only, then reorder top-level definitions, then the modification of enum value order and RuntimeDataFunctionInfo2.  This way the first two should have no functional impact (NFC), while the last contains all the functional impact.